tests/core/nodes/test_tools.py

```
@@ -0,0 +1,302 @@
+class TestToolsFromNodes:
+    """Test tools_from_nodes utility function."""
+
+    def test_single_tool_capable_node(self, bash_node):
+        """Single tool-capable node creates one tool."""
+        tools, executor = tools_from_nodes([bash_node])
+
+        assert len(tools) == 1
+        assert tools[0].name == "bash"
+        assert callable(executor)
+
+    def test_filters_non_tool_capable(self, session, bash_node):
```

Ruff (0.14.10)

Unused method argument: `session`

(ARG002)